### ISSUE TITLE
GameGuru MAX - Bug - Locked Point Light - Can't Unlock Plus Other Issues

### ISSUE THREAD

**Issue by ArgentArts** [TheGameCreators/GameGuruRepo#2410]
If you lock a Point Light in MAX, you can't unlock it (nothing shows up under Current Objects when selected to allow you to unlock it) and, even if selected, nothing shows up on the right-side panel to allow you to change it's properties:

![locked1](https://user-images.githubusercontent.com/36735507/156861938-955a14e2-d46e-417f-8bb1-fa90cd45a76b.png)

However, if you add another point light in the scene, then select the locked light, you can then access it's properties and change it's color, Light Distance, etc., but you still nothing shows up under Current Objects to allow you to unlock it:

![locked2](https://user-images.githubusercontent.com/36735507/156862060-862dba47-4f4d-43c5-9b12-61f4b9df4c78.png)

As a result, this light cannot be moved or deleted.

**Comment by ArgentArts** [comment]
And, yes, if you lock the newly added light, it, too, becomes unlockable like the others and functions in the same way. Keep doing this and you'll end up with an unlimited number of lights that you can never remove. ;)

**Comment by LeeBamberTGC** [comment]
@dllllllllllb Could be a tricky one but see how far into the rabbit hole you can go to find the WHY of this one.

**Comment by dllllllllllb** [comment]
@ArgentArts Is this still an issue? It works perfectly fine on the build I'm currently on.
If you're still getting this issue, please send me your levels .fpm file to my email: maciej@thegamecreators.com

**Comment by ArgentArts** [comment]
No, this does not work at all. If you create a light, lock it in Current Objects, click anywhere to select anything to not select the light, then I cannot re-select the light to unlock it. The only way I can unlock this light is to go to the Detailed Objects List, twirl down the Light list, select the light there, and then unlock it. That's not a good way to do this. If I have 50 lights in the scene, I won't know which light this particular one is. I would have to click on each on in that list. I should be able to at least select the light in the view port, even if it's locked. I just shouldn't be able to move it, etc., once selected. But it should show up in Current Objects so I can unlock it if I want to.

I can still send you a level, but I just did this in an empty. It's easy to duplicate:

Create an empty. Drag in a light. Click the lock button under Current Objects. Click off the light to deselect it (or press the spacebar). Now click back on the locked light. Nothing shows up in the Current Objects area, so you're not able to unlock the light this way. I'll do a video of this in a moment. :)

**Comment by ArgentArts** [comment]
https://user-images.githubusercontent.com/36735507/157682520-d25efe3b-348c-44f9-bb45-7d09d5545385.mp4

**Comment by plemsoft** [comment]
@ArgentArts It still show in "Object Tools" , it will display as read only , but you can use the "Unlock" icon :)

**Comment by ArgentArts** [comment]
Ah! My G*d. I never even saw that unlock/lock icon there. I am an idiot. :D

**Comment by dllllllllllb** [comment]
@ArgentArts If it gives you the option to lock it in there, it should also give you an option to unlock it in the same place, so the bug is valid in my opinion :] 

I was unlocking it in the top right corner, that's why I couldn't reproduce it, I'll have a look into it now 👍

**Comment by ArgentArts** [comment]
Well, the Current Objects panel does not show locked items of any kind. So, not sure that's a bug. It just doesn't consider a locked item a "current object". Because, remember, you can drag current objects from there to make duplicates of them to populate your level. So, I suppose it's not showing the locked object because, since locked, you can't do that.
